### PR TITLE
Don't publish vsix from the 'dev16' branch.

### PR DESCRIPTION
**Pull request by Pilchie**
For dogfooding purposes, we'd like to the highest version to be
the one we want people to dogfood, which for now is master.

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @dotnet/roslyn-infrastructure @jaredpar @tannergooding @jmarolf @brettfo for thoughts.

File: src/Tools/MicroBuild/publish-assets.ps1 [hunk 1]
```diff
@@ -119,37 +119,44 @@ try
     }
     popd
 
-    Write-Host "Uploading VSIX extensions..."
+    if ($branchName -neq "dev16")
```
Comment (jaredpar) [inline src/Tools/MicroBuild/publish-assets.ps1]: Won't this still cause us to release from both master and dev15.X?

Comment (Pilchie) [inline src/Tools/MicroBuild/publish-assets.ps1]: Yes, but master will have higher version numbers.

**Comment by Pilchie** [comment]
@dotnet/roslyn-infrastructure any idea how to diagnose what went wrong with https://ci.dot.net/job/dotnet_roslyn/job/dev16/job/windows_debug_vs-integration_prtest/50/? I don't see anything in the log or screenshots :(

**Comment by Pilchie** [comment]
retest windows_debug_vs-integration_prtest please

**Pull request merged**